catalogue/src/main/kotlin/com/tidal/sdk/catalogue/generated/infrastructure/URIAdapter.kt

```diff
@@ -0,0 +1,21 @@
+package com.tidal.sdk.catalogue.generated.infrastructure
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.SerialDescriptor
+import java.net.URI
+
+@Serializer(forClass = URI::class)
```

Comment:
way better than URL but I'd still argue a simple string might be the better data representation in a data class